r/R/dataset-scan.R

```diff
@@ -85,10 +85,22 @@ Scanner$create <- function(dataset,
       # To handle mutate() on Table/RecordBatch, we need to collect(as_data_frame=FALSE) now
       dataset <- dplyr::collect(dataset, as_data_frame = FALSE)
     }
+
+    proj<- c(dataset$selected_columns, dataset$temp_columns)
+    if (!is.null(projection)) {
+      # TODO: should we check names and make sure we only project once per?
+      proj <- c(proj, projection)
```

Comment:
I don't think you want to `c()` because IIUC the projection should be applied after the existing projection. One possibility is that we special case for here and only accept a character vector here, so you'd end up doing:

```suggestion
      proj <- proj[projection]
```